### PR TITLE
Split and rename DestinationElder to DataHandler

### PR DESCRIPTION
**Pull request by octol**
Rename DestinationElder to DataHandler and extract out {I,A,M}DataHandler and IDataHolder.

Part of #758

### CODE REVIEW

File: src/data_handler/adata_handler.rs [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+            Rc::clone(total_used_space),
+            init_mode,
+        )?;
+        Ok(Self {
+            id,
+            append_only_chunks,
+        })
+    }
+
+    pub(crate) fn handle_put_adata_req(
```
Comment (madadam) [inline src/data_handler/adata_handler.rs]: Another nitpick: All these methods (and the ones in the other handlers too) can be `pub(super)` as they are only called from `DataHandler` and not above that.

File: src/lib.rs [hunk 2]
```diff
@@ -32,7 +31,6 @@
     unconditional_recursion,
     unknown_lints,
     unsafe_code,
-    unused,
```
Comment (madadam) [inline src/lib.rs]: Why remove this?

File: src/lib.rs [hunk 3]
```diff
@@ -17,8 +17,7 @@
     exceeding_bitshifts,
     mutable_transmutes,
     no_mangle_const_items,
-    unknown_crate_types,
-    warnings
```
Comment (madadam) [inline src/lib.rs]: Why remove this?

File: src/vault.rs [hunk 4]
```diff
@@ -34,7 +34,7 @@ const STATE_FILENAME: &str = "state";
 enum State {
     Elder {
         src: ClientHandler,
```
Comment (madadam) [inline src/vault.rs]: Maybe also rename these fields: `src` -> `client`, `dst` -> `data` ?

File: src/vault.rs [hunk 5]
```diff
@@ -171,8 +171,8 @@ impl Vault {
         match action {
             ForwardClientRequest(message) => {
                 let requester_name = utils::rpc_elder_address(&message)?;
-                let dst_elders_address = if let Rpc::Request { ref request, .. } = message {
-                    match utils::dst_elders_address(&request) {
+                let data_handlers_address = if let Rpc::Request { ref request, .. } = message {
+                    match utils::data_handlers_address(&request) {
                         Some(address) => address,
                         None => {
                             error!("{}: Logic error - no dst address available.", self);
```
Comment (madadam) [inline src/vault.rs]: "...no data handler address available" - or something like that.

File: src/vault.rs [hunk 6]
```diff
@@ -185,10 +185,10 @@ impl Vault {
                 };
 
                 // TODO - once Routing is integrated, we'll construct the full message to send
-                //        onwards, and then if we're also part of the dst elders, we'll call that
+                //        onwards, and then if we're also part of the data handlers, we'll call that
                 //        same handler which Routing will call after receiving a message.
 
-                if self.self_is_elder_for(&dst_elders_address) {
+                if self.self_is_elder_for(&data_handlers_address) {
```
Comment (madadam) [inline src/vault.rs]: Maybe rename `self_is_elder_for` -> `self_is_handler_for` ?

File: src/lib.rs (hunk 2 above)
Comment (octol) [reply src/lib.rs]: Good catch

File: src/data_handler.rs [hunk 7]
```diff
@@ -274,11 +274,6 @@ impl DataHandler {
             //
             // ===== Coins =====
             //
```
Comment (madadam) [inline src/data_handler.rs]: You can now remove this comment (and the "Login packets" one below)

Comment (octol) [reply src/data_handler.rs]: Updated

**Pull request merged**